src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/Transmission/OtlpExporterPersistentStorageRetryTransmissionHandler.cs

```diff
@@ -0,0 +1,110 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+#nullable enable
+
+using Google.Protobuf;
+using OpenTelemetry.Exporter.OpenTelemetryProtocol.Implementation.ExportClient;
+using OpenTelemetry.PersistentStorage.Abstractions;
+using OpenTelemetry.PersistentStorage.FileSystem;
+using OpenTelemetry.Proto.Collector.Logs.V1;
+using OpenTelemetry.Proto.Collector.Metrics.V1;
+using OpenTelemetry.Proto.Collector.Trace.V1;
+
+namespace OpenTelemetry.Exporter.OpenTelemetryProtocol.Implementation.Transmission;
+
+internal sealed class OtlpExporterPersistentStorageRetryTransmissionHandler<TRequest> : OtlpExporterTransmissionHandler<TRequest>
+{
+    private readonly ManualResetEvent stopEvent = new(false);
+    private readonly Thread thread;
+    private readonly PersistentBlobProvider? persistentBlobProvider;
+    private readonly Func<byte[], TRequest>? requestFactory;
+
+    public OtlpExporterPersistentStorageRetryTransmissionHandler(IExportClient<TRequest> exportClient, Func<byte[], TRequest> requestFactory, string storagePath)
+        : base(exportClient)
+    {
+        this.persistentBlobProvider = new FileBlobProvider(storagePath);
+        this.requestFactory = requestFactory;
+
+        this.thread = new Thread(this.ProcessRetryRequestsThreadBody)
+        {
+            Name = $"OtlpExporter Persistent Retry Storage - {typeof(TRequest)}",
+            IsBackground = true,
+        };
+
+        this.thread.Start();
+    }
+
+    protected override bool OnSubmitRequestFailure(TRequest request, ExportClientResponse response)
+    {
```

Comment:
Shouldn't this be calling into ShouldRetryRequest?

Reply:
Yes, I will get this corrected.